### PR TITLE
Add domain pending renewal state

### PR DESCRIPTION
**Pull request by delputnam**
#### Changes proposed in this Pull Request

This PR adds the state to the domain assembler and uses it to show a pending renewal stat in the domain list and the domain management pages.

Screen Shot 2021-10-06 at 2 16 12 PM

Screen Shot 2021-10-06 at 2 24 08 PM

#### Testing instructions

Set the pending renewal state on a domain as specified in D67981-code.
Check the domain list and the domain management pages to make sure the pending renewal status is shown properly.
On the domain management page, make sure that the "Renew now for..." button is not shown.

Remove the renewal pending state and make sure the state shows as "Active" and the "Renew now for..." button is shown again.

### CODE REVIEW

File: client/lib/domains/resolve-domain-status.js [hunk 1]
```diff
@@ -139,6 +139,17 @@ export function resolveDomainStatus(
 			};
 
 		case domainTypes.REGISTERED:
+			if ( domain.isPendingRenewal ) {
+				const pendingRenewalMessage = translate( 'Renewal in progress' );
```
Comment (a8ci18n) [inline client/lib/domains/resolve-domain-status.js:143]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **11** times:
`translate( 'Domain renewal already in progress.' )` *ES Score: 7*

**Review by rafaelgallani — DISMISSED** [review]
Tested and confirmed that the new state is shown, but the renewal button is not hidden in this state. Other than that, everything is working as intended and LGTM 😸:  

- [X] **Domain management - List**

image

- [X] **All domains**

image

- [ ] Domain management - Details

image

**Comment by delputnam** [comment]
Eh.  I missed pushing a commit.  Fixed.

**Review by rafaelgallani — APPROVED** [review]
LGTM 😺

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6739013

Thank you @delputnam for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6739013).

**Pull request merged**